### PR TITLE
codeintel: Large-chunk reorganize autoindexing store

### PR DESCRIPTION
**Pull request by efritz**
Reorganize chunks of code to follow the broad organization in the interface definition.

## Test plan

CI.

### CODE REVIEW

File: enterprise/internal/codeintel/autoindexing/internal/store/scheduler.go [hunk 1]
```diff
@@ -164,3 +166,50 @@ WHERE
...
+
+const markRepoRevsAsProcessedQuery = `
+UPDATE codeintel_autoindex_queue
+SET processed_at = NOW()
+WHERE id = ANY(%s)
+`
+
+//
+//
+
```
Comment (eseliger) [inline enterprise/internal/codeintel/autoindexing/internal/store/scheduler.go:209]:
```suggestion
```

not sure if you want to keep that in

File: enterprise/internal/codeintel/autoindexing/internal/store/config_inference_test.go [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package store
+
+// TODO
```
Comment (eseliger) [inline enterprise/internal/codeintel/autoindexing/internal/store/config_inference_test.go:3]: is this meant to be done pre-merge?

**Review by eseliger — APPROVED** [review]

File: enterprise/internal/codeintel/autoindexing/internal/store/config_inference_test.go (hunk 2 above)
Comment (efritz) [reply enterprise/internal/codeintel/autoindexing/internal/store/config_inference_test.go:3]: No, just flagging missing tests as we do a finer-toothed comb-through.

File: enterprise/internal/codeintel/autoindexing/internal/store/scheduler.go (hunk 1 above)
Comment (efritz) [reply enterprise/internal/codeintel/autoindexing/internal/store/scheduler.go:209]: Yeah right now they separate methods and scanner/util blocks.

Comment (efritz) [reply enterprise/internal/codeintel/autoindexing/internal/store/scheduler.go:209]: Everyone hates theses so much!!

**Pull request merged**